### DISCUSSION TITLE
useFieldArray + controlled component not working

### DISCUSSION THREAD

**Discussion by sederaa** [react-hook-form/react-hook-form#3572] [Q&A] [answered]
Hi

useFieldArray with a controlled component in each row isn't working. 

CodeSandbox: https://codesandbox.io/s/react-hook-form-usefieldarray-forked-e83d5?file=/src/index.js.

1. On load, the default value ('Luo') is disappearing from the react hook form state (i.e. the data structure returned from watch()). In the CSB, see how watch() shows '{"test":[{"firstName":"Bill"}]}' without the lastName field in the defaultValues.

   However, editing last name correctly populates react hook form state.

2. When appending a second row, there is the same state issue as (1). Additionally, when deleting one of the rows, the lastName is nuked from the first (and any other) row too, but only in react hook form state (see the watch()).

In both cases the state of the uncontrolled firstName component is correct, so it's clearly an issue with using the controlled component.

Thanks
Seb

**Comment by spirit-team** [comment, marked as answer]
@sederaa  Try something like this:
https://codesandbox.io/s/react-hook-form-usefieldarray-forked-7kygg?file=/src/index.js

**Reply by sederaa** [reply]
Thanks @spirit-team, that works. 

It appears the key change is to use Controller to wrap the input.

**Reply by bluebill1049** [reply]
Thanks @spirit-team 👍
